core/src/main/kotlin/org/evomaster/core/problem/rest/RestActionBuilderV3.kt

```diff
@@ -784,4 +785,24 @@ object RestActionBuilderV3 {
         return basePath
     }
 
+    /**
+     * build a rest action based on the given [url]
+     */
+    fun buildActionBasedOnUrl(baseUrl: String, id : String, verb: HttpVerb, url: String, skipOracleChecks : Boolean) : RestCallAction?{
+
+        // if the url does not start with baseUrl (i.e., not from SUT), then there might be no point to execute this rest action
+        if (!url.startsWith(baseUrl)) return null
+
+        val uri = URI(url)
+//        Lazy.assert { "${uri.scheme}://${uri.host}:${uri.port}" == baseUrl }
+
+        val path = RestPath("${uri.scheme}://${uri.host}:${uri.port}${uri.path}".removePrefix(baseUrl.removeSuffix("/")))
+        val query : MutableList<Param> = uri.query?.split("&")?.map { q->
+            val keyValue = q.split("=")
+            Lazy.assert { keyValue.size == 2 }
```

Comment:
this is not fully correct: see [https://datatracker.ietf.org/doc/html/rfc3986#section-3.4]()

rather than crash here, should issue a warning that non-key-pairs are not currently supported.
also , could have an empty string value in the form of `key=`, ie, no char after the `=`